lib/rubocop/cop/lint/empty_file.rb

```diff
@@ -21,14 +21,24 @@ module Lint
       #   # File consisting only of comments
       #
       class EmptyFile < Base
+        extend AutoCorrector
+
         MSG = 'Empty file detected.'
 
         def on_new_investigation
-          add_global_offense(MSG) if offending?
+          return unless offending?
+
+          add_global_offense(MSG) do
+            autocorrect if autocorrect_requested?
```

Comment:
It seems weird to me that you need to check for `autocorrect_requested?`, shouldn't that be handled transparently outside of each cop?

Reply:
This is how `Lint/ScriptPermission` does it as well